blogposts/2021/better-onboarding-prevent-codebase-overwhelm.md

```diff
@@ -15,181 +15,110 @@ published: true
 
-## Onboarding now and then
+The engineering manager’s initiative plays a crucial role in tailoring onboarding to the employee. Jean recalls one engineering manager who spent an hour daily with a new hire to answer high-level questions. This hands-on, open communication led to one of the fastest onboardings he has seen, where the new engineer was able to drive a significant change in their first month on the job.
 
-In talking to engineering leaders who have been in the industry for awhile, it becomes apparent that many things have become more complex about software development over time. Software systems are serving more users, dealing with greater volumes of data, and provide more integrations and pieces of functionality than ever before. The amount of code in the world is increasing, and so has the size of the average codebase.
+Ryan Djurovich, a DevOps Manager at Xero, Australia’s largest accounting software company, takes a different tactic. He puts time on the calendar for a one-on-one, but leaves the option for individuals to cancel at their discretion. “Some people opt for a lot of sessions and come with a list of questions, and that’s how they prefer to learn. Others prefer to find information asynchronously through [services like] Slack or Confluence.”
 
-It can sometimes be tempting to yearn for the way things were when codebases were smaller and less interconnected—before the advent of distributed systems, of big complex web applications, of software permeating every company.
+Experience level also affects how new hires onboard. Junior engineers generally need more guidance and also reassurance about what is a “normal pace.” Typically senior engineers will direct their own onboarding after getting through the standard introductions. This makes it all the more important to tailor the process to each new hire so you set the right pace and level of guidance.
 
-Tooling has gotten much better and more advanced over time, but it feels like it's just catching up with the challenges of developing software at scale. You can no longer just attach a debugger to a single process to figure out what's going on in production, and you can no longer rely on local search to cover all the code that makes up your application.
+___
 
-Having a robust onboarding process, that is both standard/automated and human/supportive is as important as ever. This is especially true when you consider how quickly the number of developers in the world is growing. If we truly want to create a world where the vast majority of people contribute to the code economy, we are going to do it one onboarding at a time.
+There's a lot more to successful developer onboarding than setting up a local dev environment. We hope this advice has been helpful!
```

Comment:
@nickmyyz This still isn't the best, but I'm expecting we will have the guide or your asset to link to shortly, so we will have a better outro to include then!